### ISSUE TITLE
Broken link in navigation for `Advanced > 6. Specifications and types`

### ISSUE THREAD

**Issue by zorn** [elixirschool/school_house#151]
The link for the lesson `Advanced > 6. Specifications and types` within the navigation bar currently links to `#` but the lesson can be found if you go spelunking for it.

<https://elixirschool.com/en/lessons/advanced/typespec>

<img width="1456" alt="Screen Shot 2021-10-02 at 12 28 59 PM" src="https://user-images.githubusercontent.com/52168/135724894-5eb52c52-8904-4148-aac4-c7720a55111f.png">

**Comment by doomspork** [comment]
Thanks for the report @zorn! We will look into this

**Comment by kinson** [comment]
Yes, thanks @zorn , I also just noticed the same behavior for Metaprogramming (https://elixirschool.com/en/lessons/advanced/metaprogramming) - the page exists but the menu link is not working.

**Comment by rsmidt** [comment]
In the [_lesson_menu template](https://github.com/elixirschool/school_house/blob/a50891fc1670b9992c664df0eb2a6465602b9a9c/lib/school_house_web/templates/layout/_lesson_menu.html.heex#L156) the atom is `:typespecs` while in the [lesson config](https://github.com/elixirschool/school_house/blob/b8bf87795bb5765b4eb4209de880ac8b21a34f6e/config/lessons.exs#L35) it's `:typespec`. I can file a PR later.
